src/main/resources/documentation/domain-model-overview.md

```diff
@@ -0,0 +1,257 @@
+######Dependencies
++ FieldWorker
++ Individual
+
+######Required Fields:
++ Date startDate
++ String relationshipType
++ Individual individualA
++ Individual individualB
+
+___
+####Residency
+(child of `AuditableCollectedEntity`)
+#####What is it?
+A `Residency` models the relationship between an `Individual` and a `Location`.
+######Dependencies
++ FieldWorker 
++ Individual
++ Location
+
+######Required Fields:
++ Date startDate
++ String relationshipType
++ Individual individualA
++ Individual individualB
+
+___
+#Update Entities
+####Visit
+(child of `AuditableExtIdEntity`)
+#####What is it?
+A `Visit` models a `FieldWorker`’s follow-up visit to a particular `Location` at a particular date. `Visit`s are necessary for all “update” events that can take place within a population.
+######Dependencies
++ FieldWorker
++ Location 
+
+######Required Fields:
++ Date visitDate
++ Location location
+
+___
+####Death
+(child of `AuditableCollectedEntity`)
+#####What is it?
+A `Death` models the death of one of the `Individual`s within the area of study.
+######Dependencies
++ FieldWorker
++ Visit
++ Location 
+
+######Required Fields:
++ Date deathDate
++ Visit visit
++ Individual individual
+
+___
+####InMigration
+(child of `AuditableCollectedEntity`)
+#####What is it?
+An `InMigration` represents a migration between two `Location`s where the destination
+`Location` is still inside the area of study. The origin `Location` can be within or outside the area of study and is model by the migrationType value of internal or external respectively.
+Dependencies
+######FieldWorker
++ Visit
++ Residency
++ Individual
+
+######Required Fields:
++ Visit visit
++ Date migrationDate
++ String migrationType
++ Residency residency
++ Individual individual
+
+___
+####OutMigration
+(child of `AuditableCollectedEntity`)
+#####What is it?
+An `OutMigration` models the migration of an `Individual` to outside the area of study.
+######Dependencies
++ FieldWorker
++ Visit
++ Residency
++ Individual 
+ 
+######Required Fields:
++ Visit visit
++ Date migrationDate
++ Residency residency
++ Individual individual
+
+___
+####PregnancyObservation
+(child of `AuditableCollectedEntity`)
+#####What is it?
+A `PregnancyObservation` models a `FieldWorker`’s observation of a pregnant `Individual`.
+######Dependencies
++ FieldWorker
++ Visit
++ Individual
+
+######Required Fields:
++ Visit visit
++ Date expectedDeliveryDate
++ Date pregnancyDate
++ Individual mother
+
+___
+####PregnancyOutcome
+(child of AuditableCollectedEntity)
+#####What is it?
+A `PregnancyOutcome` models the outcome of an `Individual`’s pregnancy.
+######Dependencies
++ FieldWorker
++ Visit
++ Individual
+
+######Required Fields:
++ Visit visit
++ Date outcomeDate
++ int childrenBorn
++ Individual mother
+
+___
+####PregnancyResult
+(child of AuditableCollectedEntity)
+#####What is it?
+A `PregnancyResult` models the separate results of a larger `PregnancyOutcome` for an `Individual`’s pregnancy. 

```

Comment:
For example, each child in a set of twins would have her own PregnancyResult.

Outside the scope of this documentation:
PregnancyOutcome and PregnancyResult are confusing!  I like how the reference data model paper models the Pregnancy itself as an episode instead of having separate event tables for observation and outcome.